hosted/aks/helper/helper_cluster.go

```diff
@@ -303,14 +307,12 @@ func defaultAKS(client *rancher.Client, cloudCredentialID, region string) (defau
 		return
 	}
 
-	maxValue := helpers.HighestK8sVersionSupportedByUI(client)
-
-	// Iterate in the reverse order to get the highest version
+	maxValue := helpers.HighestK8sMinorVersionSupportedByUI(client)
+	// Compare with value obtained from helpers.HighestK8sMinorVersionSupportedByUI to get the second-highest version so that upgrade tests can be done
 	// We obtain the value similar to UI; ref: https://github.com/rancher/ui/blob/master/lib/shared/addon/components/cluster-driver/driver-azureaks/component.js#L140
-	for i := len(versions) - 1; i >= 0; i-- {
-		if strings.Contains(versions[i], maxValue) {
-			defaultAKS = versions[i]
```

Comment:
Yes, I thought that was the intention. Since there are 2 variants of a minor version supported at a given time(for 1.28, there is 1.28.3 and 1.28.5), using the second highest variant can still allow us to test upgrade cases.

Reply:
As per our slack discussion, we will use the default UI AKS version for p0NodeChecks, but a variant of second-highest minor k8s version for p0K8sUpgradeChecks, so upgrade tests 1.27 -> 1.28